ccm_web/server/localTypes/express-session/index.d.ts

```diff
@@ -4,6 +4,9 @@ declare module 'express-session' {
   interface CustomData {
     ltiKey: string
     userLoginId: string
+    courseId: number
+    roles: string[]
```

Comment:
We could do a little more research on what role types are possible and be more specific with the type. I'm thinking something like this, or an enum:

```
type CourseRoleType = 'TeacherEnrollment' | 'StudentEnrollment' ...
```

Also, if these roles are context specific, maybe we should the data shape here somewhat, maybe do something like this:
```
course: {
  id: number
  roles: CourseRoleType[]
}
```

Reply:
I would think we want to at expect multiple roles?

```
if (roles.contains(TEACHER_ROLE)) {
   userFeatures.concat(TEACHER_FEATURES)
}
if (roles.contains(ADMIN_ROLE) {
  userFeatures.concat(ADMIN_FEATURES)
}
...
   
```

Reply:
Yes, a user can have multiple roles but if he is admin then we just show then all features

Reply:
Okay, @pushyamig, so you want to kick the can on specific types down the road? I'm okay with that. What do you think about the `course` nesting?

Reply:
@ssciolla I changed course object you [suggested](https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/114/commits/8b7d6d2f7faeefecfef6f870de796d84cb321347). It was better way to do it